### PR TITLE
Add rollout upgrade docs page

### PR DESCRIPTION
**Pull request by bschimke95**
Docs that describe how to upgrade an HA `k8s-snap` cluster in CAPI.

### CODE REVIEW

File: docs/src/_parts/capi_infra_providers/aws.md
Comment (bschimke95) [inline docs/src/_parts/capi_infra_providers/aws.md:1]: Seems like I accidentally committed this file some PRs ago

File: docs/src/capi/howto/rollout-upgrades.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+Also, verify that the workload cluster runs on the expected Kubernetes version
+
+```
+kubectl --kubeconfig c1-kubeconfig.yaml get nodes -o wide
+```
+
+## Update the CK8sControlPlane resource
+
+The first step in upgrading the control plane is to update the CK8sControlPlane
+resource with the new Kubernetes version.
```
Comment (mateoflorido) [inline docs/src/capi/howto/rollout-upgrades.md:41]:
```suggestion
The first step in upgrading the control plane is to update the CK8sControlPlane
resource with the new Kubernetes version. In this example, the control plane
is called `c1-control-plane`.
```
Wdyt?

Comment (bschimke95) [reply docs/src/capi/howto/rollout-upgrades.md]: sounds good, thanks!

File: docs/src/capi/howto/rollout-upgrades.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+kubectl --kubeconfig c1-kubeconfig.yaml get nodes -o wide
+```
+
+The machines will be replaces one after each other until all machines run on
+the desired version.
+
+## Update the MachineDeployment resource
+
+After upgrading the control plane, proceed with upgrading the worker nodes
+in a similar fashion by updating the `MachineDeployment` resource.
```
[the remark this answers is not in the document]
Comment (bschimke95) [reply docs/src/capi/howto/rollout-upgrades.md]: thanks!

File: docs/src/capi/howto/external-etcd.md [hunk 3]
```diff
@@ -116,10 +116,5 @@ kubectl create -f peaches.yaml
 To check the status of the cluster, run:
 
 ```
-clusterctl describe cluster peaches 
+clusterctl describe cluster peaches
 ```
-
-<!-- LINKS -->
```
Comment (louiseschmidtgen) [inline docs/src/capi/howto/external-etcd.md:122]: Did you remove these by accident?

File: docs/src/capi/howto/rollout-upgrades.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+- A Kubernetes management cluster with Cluster API and providers installed
+  and configured.
+- A target workload cluster managed by CAPI.
+- `kubectl` installed and configured to access your management cluster.
+- The workload cluster kubeconfig. We will refer to it as `c1-kubeconfig.yaml`
+  in the following steps.
+
+Please refer to the [getting-started guide][getting-started] for instructions
+on how to set this up.
```
Comment (louiseschmidtgen) [inline docs/src/capi/howto/rollout-upgrades.md:19]:
```suggestion
Please refer to the [getting-started guide][getting-started] for further details on the required setup.
```

File: docs/src/capi/howto/rollout-upgrades.md [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+In the following steps we will refer to the workload cluster as `c1` and its
+kubeconfig as `c1-kubeconfig.yaml`.
+
+## Check the current cluster status
+
+Before starting the upgrade, ensure your management cluster is in a healthy
```
Comment (louiseschmidtgen) [inline docs/src/capi/howto/rollout-upgrades.md:25]:
```suggestion
Prior to the upgrade, ensure that the management cluster is in a healthy
```

File: docs/src/capi/howto/rollout-upgrades.md [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+state.
+
+```
+kubectl get nodes -o wide
+```
+
+Also, verify that the workload cluster runs on the expected Kubernetes version
```
Comment (louiseschmidtgen) [inline docs/src/capi/howto/rollout-upgrades.md:32]:
```suggestion
Confirm the Kubernetes version of the workload cluster is as expected:
```

File: docs/src/capi/howto/rollout-upgrades.md [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+
+```
+kubectl --kubeconfig c1-kubeconfig.yaml get nodes -o wide
+```
+
```
Comment (louiseschmidtgen) [inline docs/src/capi/howto/rollout-upgrades.md:37]: We could add a note here to tell people to only update minor versions.

File: docs/src/capi/howto/rollout-upgrades.md [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+## Update the CK8sControlPlane resource
```
Comment (louiseschmidtgen) [inline docs/src/capi/howto/rollout-upgrades.md:38]:
```suggestion
## Update the Control Plane
```
Let's keep headers simplified.

File: docs/src/capi/howto/rollout-upgrades.md [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+
+The first step in upgrading the control plane is to update the CK8sControlPlane
```
Comment (louiseschmidtgen) [inline docs/src/capi/howto/rollout-upgrades.md:40]:
```suggestion
In this first step, update the CK8sControlPlane
```

File: docs/src/capi/howto/rollout-upgrades.md [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+kubectl get ck8scontrolplane c1-control-plane -w
+```
+
+Inspect the current machines with
+
+```
+kubectl --kubeconfig c1-kubeconfig.yaml get nodes -o wide
+```
+
+The machines will be replaces one after each other until all machines run on
```
Comment (eaudetcobello) [inline docs/src/capi/howto/rollout-upgrades.md:72]:
```suggestion
The machines will be replaced in turn until all machines run on
```

File: docs/src/capi/howto/rollout-upgrades.md [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+the desired version.
+
+## Update the MachineDeployment resource
+
+After upgrading the control plane, proceed with upgrading the worker nodes
+in a similar fashion by updating the `MachineDeployment` resource. For
```
Comment (eaudetcobello) [inline docs/src/capi/howto/rollout-upgrades.md:78]:
```suggestion
by updating the `MachineDeployment` resource. For
```

**Review by eaudetcobello — CHANGES_REQUESTED** [review]

File: docs/src/capi/howto/rollout-upgrades.md [hunk 12]
```diff
@@ -0,0 +1,120 @@
...
+  template:
+    spec:
+      version: v1.30.3
+```
+
+Save and exit the editor.
+
+## Monitor the worker node upgrade
+
+Similar to the control-planes, you can monitor the upgrade with
```
Comment (louiseschmidtgen) [inline docs/src/capi/howto/rollout-upgrades.md:99]:
```suggestion
Just like with the control planes, monitor the upgrade using:
```

File: docs/src/capi/howto/rollout-upgrades.md [hunk 13]
```diff
@@ -0,0 +1,120 @@
...
+
+```
+kubectl get machinedeployment c1-worker-md
+```
+
+## Verify the upgrade
```
Comment (louiseschmidtgen) [inline docs/src/capi/howto/rollout-upgrades.md:105]:
```suggestion
## Verify the Kubernetes upgrade
```

File: docs/src/capi/howto/rollout-upgrades.md [hunk 14]
```diff
@@ -0,0 +1,120 @@
...
+
+Finally, verify that all nodes are running the new version and are healthy:
+
+```
+kubectl --kubeconfig c1-kubeconfig.yaml get nodes -o wide
+```
+
+and that no old machines are left behind:
```
Comment (louiseschmidtgen) [inline docs/src/capi/howto/rollout-upgrades.md:113]:
```suggestion
As a last step, ensure that no old machines are left behind:
```

**Review by louiseschmidtgen — COMMENTED** [review]
Great work @bschimke95! I've added a couple of suggestions to improve the flow.

File: docs/src/capi/howto/external-etcd.md (hunk 3 above)
Comment (bschimke95) [reply docs/src/capi/howto/external-etcd.md:122]: lol, no idea how I pulled that one off

File: docs/src/capi/howto/rollout-upgrades.md [hunk 15]
```diff
@@ -0,0 +1,120 @@
...
+```
+
+The machines will be replaces one after each other until all machines run on
+the desired version.
+
+## Update the MachineDeployment resource
+
+After upgrading the control plane, proceed with upgrading the worker nodes
+in a similar fashion by updating the `MachineDeployment` resource. For
+instance, we will be updating the `c1-worker-md`.
```
[the remark this answers is not in the document]
Comment (bschimke95) [reply docs/src/capi/howto/rollout-upgrades.md:79]: machine-deployment

File: docs/src/capi/howto/rollout-upgrades.md (hunk 7 above)
Comment (bschimke95) [reply docs/src/capi/howto/rollout-upgrades.md:37]: done - thanks for all the great suggestions!

File: docs/src/capi/howto/external-etcd.md (hunk 3 above)
Comment (louiseschmidtgen) [reply docs/src/capi/howto/external-etcd.md:122]: Maybe save without formatting?

File: docs/src/capi/howto/rollout-upgrades.md [hunk 16]
```diff
@@ -0,0 +1,124 @@
...
+```
+
+Replace the `spec.version` field with the new Kubernetes version.
+
+```yaml
+spec:
+  version: v1.30.3
+```
+
+Please safe your changes.
```
Comment (evilnick) [inline docs/src/capi/howto/rollout-upgrades.md:59]:
```suggestion
Please save your changes.
```

File: docs/src/capi/howto/rollout-upgrades.md [hunk 17]
```diff
@@ -0,0 +1,124 @@
...
+To inspect the current machines, execute:
+
+```
+kubectl --kubeconfig c1-kubeconfig.yaml get nodes -o wide
+```
+
+The machines will be replaced in turn until all machines run on
+the desired version.
+
+## Update the Worker nodes
```
Comment (evilnick) [inline docs/src/capi/howto/rollout-upgrades.md:79]:
```suggestion
## Update the worker nodes
```

File: docs/src/capi/howto/rollout-upgrades.md [hunk 18]
```diff
@@ -0,0 +1,124 @@
...
+Kubernetes version.
+
+```yaml
+spec:
+  template:
+    spec:
+      version: v1.30.3
+```
+
+Please safe your changes.
```
Comment (evilnick) [inline docs/src/capi/howto/rollout-upgrades.md:99]:
```suggestion
Please save your changes.
```

File: docs/src/capi/howto/rollout-upgrades.md [hunk 19]
```diff
@@ -0,0 +1,124 @@
...
+
+```
+kubectl --kubeconfig c1-kubeconfig.yaml get nodes -o wide
+```
+
+```{note} For rollout upgrades, only the minor version should be updated.
+```
+<!-- TODO(ben): add reference to in-place upgrades once we have those docs. -->
+
+## Update the Control Plane
```
Comment (evilnick) [inline docs/src/capi/howto/rollout-upgrades.md:42]:
```suggestion
## Update the control plane
```

File: docs/src/capi/howto/rollout-upgrades.md (hunk 16 above)
Comment (louiseschmidtgen) [reply docs/src/capi/howto/rollout-upgrades.md:59]: Thanks!

File: docs/src/capi/howto/external-etcd.md (hunk 3 above)
Comment (louiseschmidtgen) [reply docs/src/capi/howto/external-etcd.md:122]: Supi danke!

**Review by louiseschmidtgen — APPROVED** [review]
LGTM

**Review by eaudetcobello — APPROVED** [review]

**Pull request merged**